flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -310,8 +310,16 @@ fi
 INTERNAL_HADOOP_CLASSPATHS="${HADOOP_CLASSPATH}:${HADOOP_CONF_DIR}:${YARN_CONF_DIR}"
 
 if [ -n "${HBASE_CONF_DIR}" ]; then
-    # Setup the HBase classpath.
-    INTERNAL_HADOOP_CLASSPATHS="${INTERNAL_HADOOP_CLASSPATHS}:`hbase classpath`"
+    # Search hbase command in PATH.
+    HBASE_IN_PATH=$(PATH="${HBASE_HOME}/bin:$PATH" \
+        which hbase 2>/dev/null)
+    # Whether the hbase command is found.
+    if [ -f "${HBASE_IN_PATH}" ]; then
+        # The hbase command is found, then setup the HBase classpath.
+        INTERNAL_HADOOP_CLASSPATHS="${INTERNAL_HADOOP_CLASSPATHS}:`${HBASE_IN_PATH} classpath`"
+    else
+        echo "The HBASE_IN_PATH is setted but can't setup HBase classpath. Please verify hbase command exists."
```

Comment:
Perhaps more like "HBASE_CONF_DIR=\"${HBASE_CONF_DIR}\" is set but 'hbase' command was not found in \"${HBASE_PATH}\" so classpath could not be updated."? I assumed HBASE_PATH="${HBASE_HOME}/bin:$PATH" as a local variable.